packages/url_launcher/url_launcher_platform_interface/lib/link.dart

```diff
@@ -80,10 +80,13 @@ Future<ByteData> pushRouteNameToFramework(
   String routeName, {
   @visibleForTesting bool debugForceRouter = false,
 }) {
+  if (window.onPlatformMessage == null) {
+    return Future<ByteData>.value(null);
+  }
   final Completer<ByteData> completer = Completer<ByteData>();
   if (debugForceRouter || _hasRouter(context)) {
     SystemNavigator.routeInformationUpdated(location: routeName);
-    window.onPlatformMessage(
+    window.onPlatformMessage!(
```

Comment:
To avoid repeatedly adding `!` you can do:
```dart
final PlatformMessageCallback? onPlatformMessage = window.onPlatformMessage;
if (onPlatformMessage == null) {
  return Future<ByteData>.value(null);
}
// After this point, the analyzer knows that the local variable `onPlatformMessage` is
// not null, so there's no need to add ! anymore.
```

Reply:
good suggestion! changed